### ISSUE TITLE
Unexpected request redirect behaviour in kubernetes

### ISSUE THREAD

**Issue by octopusx** [containous/traefik#3589]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Deployed Traefik 1.6.3 to our organisation kubernetes cluster to use it for ingress. Trying to replace Nginx.
Some of our services require redirects and this is configured as annotations in their ingress kubernetes manifests.

When we did this using Nginx the following annotation did the job:
```
 apiVersion: extensions/v1beta1
 kind: Ingress
 
 metadata:
   annotations:
     nginx.ingress.kubernetes.io/app-root: /folder/index.html
```
Traefik has a similar annotation available as described [here](https://docs.traefik.io/configuration/backends/kubernetes/):
```
 apiVersion: extensions/v1beta1
 kind: Ingress
 
 metadata:
   annotations:
     traefik.ingress.kubernetes.io/app-root: /folder/index.html
```
We expected it to work in a similar fashion, as is indicated by the documentation, however the requests were not forwarded.

Much to our surprise after an amount of testing we found that enabling the ssl-redirect flag solves the problem:
```
 apiVersion: extensions/v1beta1
 kind: Ingress
 
 metadata:
   annotations:
     traefik.ingress.kubernetes.io/ssl-redirect: "true"
     traefik.ingress.kubernetes.io/app-root: /folder/index.html
```

The issue here is that either this is unexpected behaviour (that redirection doesn't work without the extra flag) or it is intended and therefore documentation should be updated accordingly.

I tried to provide an adequate level of detail without posting internal company data but please ask if you need more specific information or examples of any of the points mentioned in the issue.

### What did you expect to see?

Executing the following: `curl --verbose -L -k --header 'Host: target.com' 'http://our.traefik.deployment.com'` should get us:
```
[...]
GET / HTTP/1.1
[...]
HTTP/1.1 301 Moved Permanently
[...]
GET /folder/index.html HTTP/1.1
[...]
HTTP/1.1 200 OK
[...]
```

### What did you see instead?

Executing the following: `curl --verbose -L -k --header 'Host: target.com' 'http://our.traefik.deployment.com'` gets us:
```
[...]
GET / HTTP/1.1
[...]
HTTP/1.1 503 Service Unavailable
[...]
```
### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.6.3
Codename:     tetedemoine
Go version:   go1.10.2
Built:        2018-06-05_03:29:01PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Traefik config:
```toml
logLevel = "DEBUG"
defaultEntryPoints = ["http"]

[entryPoints]
  [entryPoints.http]
  address = ":80"

[ping]
entryPoint = "http"

[kubernetes]
namespaces = ["default", "kube-system"]

[web]
address = ":8080"

[api]
entryPoint = "traefik"
dashboard = true

[lifecycle]
gracetimeout = "80s"
requestacceptgracetimeout = "40s"
```

Kubernetes version:
```
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.3+coreos.0", GitCommit:"f1b890dbbf11abe58280b3ffe17d67749f5ae70e", GitTreeState:"clean", BuildDate:"2018-05-21T17:27:17Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

Thank you for reading!

**Comment by dtomcej** [comment]
Confirmed. According to the validation from (https://kubernetes.github.io/ingress-nginx/examples/rewrite/README/#app-root), we are not providing the same behaviour.

**Comment by traefiker** [comment]
Closed by #3592.
